### PR TITLE
feat: add new model api provider: Novita AI

### PR DESCRIPTION
**Pull request by jasonhp**

### CODE REVIEW

**Comment by reid41** [comment]
seems some indent issues, could you please help to fix it?

File: s-pilot [hunk 1]
```diff
@@ -264,9 +273,12 @@ fetch_and_cache_models() {
     elif [[ "$USE_API" == "localai" ]]; then
 		models_response=$(fetch_model_from_localai)
         echo $models_response | jq -r -C '.data[] | {id, object}' > "$cache_file"
-	elif [[ "$USE_API" == "mistralai" ]]; then
+	  elif [[ "$USE_API" == "mistralai" ]]; then
 		models_response=$(fetch_model_from_mistralai)
-	 	echo $models_response | jq -r -C '.data[] | {id, owned_by, created}'  > "$cache_file"
+	 	    echo $models_response | jq -r -C '.data[] | {id, owned_by, created}'  > "$cache_file"
+    elif [[ "$USE_API" == "novita" ]]; then
```
Comment (reid41) [inline s-pilot]: indent

File: s-pilot [hunk 2]
```diff
@@ -264,9 +273,12 @@ fetch_and_cache_models() {
     elif [[ "$USE_API" == "localai" ]]; then
 		models_response=$(fetch_model_from_localai)
         echo $models_response | jq -r -C '.data[] | {id, object}' > "$cache_file"
-	elif [[ "$USE_API" == "mistralai" ]]; then
+	  elif [[ "$USE_API" == "mistralai" ]]; then
```
Comment (reid41) [inline s-pilot]: indent

File: s-pilot [hunk 3]
```diff
@@ -526,6 +538,8 @@ code_chat_function(){
 				response_data=$(echo $response | jq -r '.content[0].text')
 			elif [[ "$USE_API" == "moonshot" ]]; then
 				response_data=$(echo $response | jq -r '.choices[].message.content')
+      elif [[ "$USE_API" == "novita" ]]; then
+        response_data=$(echo $response | jq -r '.choices[].message.content')
 			fi
```
Comment (reid41) [inline s-pilot:543]: indent

File: s-pilot [hunk 4]
```diff
@@ -760,6 +774,9 @@ while [[ "$#" -gt 0 ]]; do
 		elif [[ "$USE_API" == "moonshot" ]]; then
 			update_config "$2" "MODEL_MOONSHOT"
 			check_the_setting "moonshot model" "MODEL_MOONSHOT"
+    elif [[ "$USE_API" == "novita" ]]; then
+      update_config "$2" "MODEL_NOVITA"
+      check_the_setting "novita model" "MODEL_NOVITA"
 		fi
```
Comment (reid41) [inline s-pilot:780]: indent

File: s-pilot [hunk 5]
```diff
@@ -929,6 +946,15 @@ while $running; do
...
 			] | .[]' 
+    elif [[ "$USE_API" == "novita" ]]; then
+			models_response=$(fetch_model_from_novitaai)
+			model_data=$(echo $models_response | jq -r -C '.data[] | select(.id=="'"${prompt#*model:}"'")')
+			# stop the spinner
+			cat ${SPINNER_SIGNAL_FILE} >/dev/null 2>&1
+			echo "1" > ${SPINNER_SIGNAL_FILE}
+			wait $dynamic_spinner_pid
+			echo -e "$OVERWRITE_PROCESSING_LINE"
+			echo -e "${SHELLPILOT_CYAN_LABEL}Complete details for model: ${prompt#*model:}\n ${model_data}"
```
Comment (reid41) [inline s-pilot:957]: indent

File: s-pilot [hunk 6]
```diff
@@ -968,6 +994,8 @@ while $running; do
 			response_data=$(echo $response | jq -r '.choices[].message.content')
 		elif [[ "$USE_API" == "anthropic" ]]; then
 			response_data=$(echo $response | jq -r '.content[0].text')
+    elif [[ "$USE_API" == "novita" ]]; then
+      response_data=$(echo $response | jq -r '.choices[].message.content')
 		fi
```
Comment (reid41) [inline s-pilot:999]: indent

File: s-pilot [hunk 7]
```diff
@@ -1043,6 +1071,8 @@ while $running; do
...
```
Comment (reid41) [inline s-pilot:1076]: indent

File: s-pilot (hunk 3 above)
Comment (jasonhp) [reply s-pilot:543]: Should I use "/t" instead of spaces to indent here?

**Comment by jasonhp** [comment]
I think I fixed those indent problems. Is this okay?

**Comment by reid41** [comment]
yes, LGTM,  did you test in your side?

**Comment by jasonhp** [comment]
> yes, LGTM, did you test in your side?

Yes. I tested and it worked well.

**Pull request merged**